File: batchdelete.py
from pymongo import MongoClient
from datetime import datetime, timedelta, timezone
import time


client = MongoClient('mongodb+srv://root1:1234@x.kisec.mongodb.net/')
db = client['sample']
collection = db['minimax_ttl1']



start_time = datetime(2025, 5, 15, 11, 42, 00, tzinfo=timezone.utc)
end_time = datetime(2025, 5, 15, 12, 42, 00, tzinfo=timezone.utc)

# 步长
step = timedelta(seconds=1)

current_time = start_time
total_deleted = 0

print(f"开始按时间批量删除文档，时间范围: {start_time} 到 {end_time}")

while current_time <= end_time:

    delete_result = collection.delete_many({"expired_at_date": {"$lt": current_time}})
    

    deleted_count = delete_result.deleted_count
    total_deleted += deleted_count

    local_time = time.localtime()
    formatted = time.strftime("%Y-%m-%d %H:%M:%S", local_time)
    print(f"{formatted} 删除 {current_time} 之前的文档: {deleted_count} 条")
    

    current_time += step

    
    time.sleep(0.7)

print(f"删除操作完成，共删除 {total_deleted} 条文档")


client.close()

File: batchdelete.js

const dbName = 'test';
const collectionName = 'test';


const startTime = new Date("2025-05-15T11:42:00.000Z");
const endTime = new Date("2025-05-15T12:42:00.000Z");


const stepSeconds = 1;
const sleepMilliseconds = 700; 



const db = db.getSiblingDB(dbName);
const collection = db.getCollection(collectionName);

let currentTime = new Date(startTime.getTime());
let totalDeleted = 0;

print(`开始按时间批量删除文档，时间范围: ${startTime.toISOString()} 到 ${endTime.toISOString()}`);

while (currentTime <= endTime) {
    const deleteResult = collection.deleteMany({ "expired_at_date": { "$lt": currentTime } });

    const deletedCount = deleteResult.deletedCount;
    totalDeleted += deletedCount;

    const localTimeFormatted = new Date().toLocaleString();
    
    print(`${localTimeFormatted} | 删除 ${currentTime.toISOString()} 之前的文档: ${deletedCount} 条`);

    currentTime.setSeconds(currentTime.getSeconds() + stepSeconds);

    sleep(sleepMilliseconds);
}

print(`\n删除操作完成，共删除 ${totalDeleted} 条文档`);




